### DISCUSSION TITLE
Show and tell: Deploying NiceGUI app in Azure WebApp with Authentication

### DISCUSSION THREAD

**Discussion by Rjdrenth** [zauberzeug/nicegui#3624] [Show and tell]
Hi everybody,

Recently I have been working on integrating Azure Entra Authentication into my NiceGUI app that I created for internal use within my company. As we already use Microsoft accounts within our company, Azure Entra was used to authenticate users and limit access. 
It was quite a journey, which might have been due to a lack of experience in this area, so I decided to share my solution.

It is not perfect, e.g. I would have liked to incorporate some catch-all middleware as demonstrated [in the NiceGUI authentication example](https://github.com/zauberzeug/nicegui/blob/main/examples/authentication/main.py), but I ended up with a working solution. 

I would like to share the results with anyone who is interested with a minimal example, [which can be viewed here](https://github.com/Rjdrenth/nicegui-azure-entra-auth/blob/main/nicegui_azure_entra_auth/minimal_nicegui_entra_auth_example.py). 

It only covers the NiceGUI side of things and does not cover all the Azure-side hoops you need to go through, such as registering an Entra application and whatnot. Nevertheless, I hope it is useful to somebody. Any feedback is welcome.

**Comment by alikalik9** [comment]
Thank you so much, I am looking for exactly something like this right now. I also would be intrested in the azure part, as I dont have much experience with hosting on azure + EntraID

**Comment by pimswr** [comment]
Thank you very much! I had to implement it and you did it perfectly. It worked directly out of the box. I just hat to set the login parameters. Great job, thanks again.

**Reply by Rjdrenth** [reply]
Glad to hear it!

**Reply by Rjdrenth** [reply]
You might be interested to know that I updated the example slightly concerning obtaining public keys from Azure to verify the JWT tokens.
